### PR TITLE
Fix for FIPS startup

### LINKED ISSUE #358: Add option to specify PKCS12 algorithm in pre-start.erb
On a FIPS enabled stemcell, pre-start.erb fails under a Bionic based stemcell.

The issue is that the current Ubuntu Bionic based stemcell uses an older version of OpenSSL, which it is required to do to run in FIPS mode.  FIPS modules are validated against specific versions, and it is not possible to patch to a newer version without loosing validation.  There is a known bug in the validated version that when using the openssl pkcs12 command, if the option -certpbe is not used to specify an algorithm, openssl will default to a non-FIPS algorithm. 

The pre-start.erb script contains the function insert_ssl_cert which uses openssl pkcs12 to export the SSL cert.  Because -certpbe is not specified, it is failing on a FIPS enabled stemcell.  However, we have validated that passing a supported algorithm such as PBE-SHA1-3DES, everything works as expected.

I'd like the script to either check for a FIPS enabled kernel and select the correct algorithm, or the ability to specify which algorithm the script should use.

### PR DESCRIPTION
**Pull request by tack-sap**
Addresses #358

### CODE REVIEW

**Comment by strehle** [comment]
here I cannot judge if this is a regression for others / older installations 

@torsten-sap  @friegger  we need someone with BOSH expierence to verify that this change does not break existing landscapes

**Review by strehle — APPROVED** [review]
if you have checked that this option does not break start on non-/old - VMs thank ok from my side

File: jobs/uaa/templates/bin/pre-start.erb [hunk 1]
```diff
@@ -123,7 +123,7 @@ function process_certs {
 function insert_ssl_cert {
     log "Installing Server SSL certificate"
 
-    openssl pkcs12 -export -name uaa_ssl_cert \
+    openssl pkcs12 -export -certpbe PBE-SHA1-3DES -name uaa_ssl_cert \
```
Comment (rkoster) [inline jobs/uaa/templates/bin/pre-start.erb:126]: Can we expose this through a bosh property, instead of hard coding this change?

**Review by rkoster — CHANGES_REQUESTED** [review]

**Comment by pburkholder** [comment]
SHA1 and 3DES raises hackles here. Aren't there other ciphers/hashes that aren't so tied to the 20th century? 

SHA1 was deprecated nearly a decade ago.

File: jobs/uaa/templates/bin/pre-start.erb (hunk 1 above)
Comment (strehle) [reply jobs/uaa/templates/bin/pre-start.erb:126]: ok so I agree, please use a parameter

**Comment by strehle** [comment]
> SHA1 was deprecated nearly a decade ago.

The algorithm is used for p12 creation, but the password is passed 3 lines later, so this parameter is not needed to increase or change security level, but simply on a FIPS compliant image the openssl version requires such a parameter. We convert key files into p12 to pass the key material to tomcat configuration.

File: jobs/uaa/templates/bin/pre-start.erb (hunk 1 above)
Comment (toabctl) [reply jobs/uaa/templates/bin/pre-start.erb:126]: you could also check (`cat /proc/sys/crypto/fips_enabled`) if FIPS is enabled and only add that parameter in the FIPS case.

Comment (toabctl) [reply jobs/uaa/templates/bin/pre-start.erb:126]: https://github.com/cloudfoundry/uaa-release/pull/407 is an alternative using the fix I suggested here.

**Pull request closed without merging**